modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/sql/ClientSqlExecuteRequest.java

```diff
@@ -76,23 +81,29 @@ public static CompletableFuture<Void> process(
 
         return session
                 .executeAsync(tx, statement, arguments)
-                .thenCompose(asyncResultSet -> writeResultSetAsync(out, resources, asyncResultSet, session));
+                .thenCompose(asyncResultSet -> writeResultSetAsync(out, resources, asyncResultSet, session, metrics));
     }
 
     private static CompletionStage<Void> writeResultSetAsync(
             ClientMessagePacker out,
             ClientResourceRegistry resources,
             AsyncResultSet asyncResultSet,
-            Session session) {
+            Session session,
+            ClientHandlerMetricSource metrics) {
         boolean hasResource = asyncResultSet.hasRowSet() && asyncResultSet.hasMorePages();
 
         if (hasResource) {
             try {
-                var clientResultSet = new ClientSqlResultSet(asyncResultSet, session);
+                metrics.cursorsActiveIncrement();
+
+                var clientResultSet = new ClientSqlResultSet(asyncResultSet, session, metrics);
 
                 ClientResource resource = new ClientResource(
                         clientResultSet,
-                        () -> clientResultSet.closeAsync().join());
+                        () -> {
+                            clientResultSet.closeAsync();
+                            metrics.cursorsActiveDecrement();
```

Comment:
But there is already `metrics.cursorsActiveDecrement()` in `clientResultSet.closeAsync()`.

Reply:
Good catch, fixed. This is a leftover from initial implementation.